tests/Properties.hs

```diff
@@ -93,6 +96,20 @@ prop_lines_lazy2 =
 
 prop_strip x = C.strip x == (C.dropSpace . C.reverse . C.dropSpace . C.reverse) x
 
+prop_toConstr :: P.ByteString -> Bool
+prop_toConstr bs = "pack" == ((showConstr  . toConstr) bs)
+
+prop_gshow_equal :: P.ByteString -> Bool
+prop_gshow_equal b = read_bs b == read_string b
+    where
+        read_bs :: P.ByteString -> [(P.ByteString, String)]
+        read_bs = gread . gshow
+        read_string :: P.ByteString -> [(P.ByteString, String)]
+        read_string = gread . Char8.unpack
```

Comment:
I've had some trouble understanding this test, because it seems to me that `Char8.unpack` and `gshow @ByteString` would behave very differently.

Now I suspect that this test is passing "trivially" because `gread` is always returning `[]`:

```
ghci> gread @ByteString "ABC"
[]
ghci> gread @ByteString (gshow @ByteString "ABC")
[]
```